src/dataService/tagDataService.ts

```diff
@@ -418,6 +427,41 @@ export class TagDataService {
     return await this.savedItemService.getSavedItemById(savedItemId);
   }
 
+  /**
+   * Replaces all tags associated with a given savedItem id
+   * @param savedItemTagsInput
+   */
+  public async replaceSavedItemTags(
+    savedItemTagsInput: SavedItemTagsInput[]
+  ): Promise<SavedItem[]> {
+    const tagInputs: TagCreateInput[] = savedItemTagsInput.reduce(
+      (inputs, tagInput) => {
+        const tagInputs = tagInput.tags.map((tag) => ({
+          savedItemId: tagInput.savedItemId,
+          name: tag,
+        }));
+
+        inputs.push(...tagInputs);
+        return inputs;
+      },
+      []
+    );
+
+    const savedItemIds = savedItemTagsInput.map((input) => input.savedItemId);
+
+    await this.insertTags(tagInputs, async (trx) => {
+      await Promise.all(
+        savedItemIds.map(async (id) => {
+          await this.deleteTagsByItemId(id).transacting(trx);
```

Comment:
nit: from single responsibility principle perspective, thinking why `insertTags` need to handle the callback. would it be better to let the calling function to handle these logic i.e call `await deleteTagsByItemId` first and then do `await insertTags`

Reply:
It would be better but I wanted to tie into the transaction for insert tags for the delete as well. I'll try something else here, I'll ping when I update